src/XR/webXRSessionManager.ts

```diff
@@ -332,6 +332,13 @@ export class WebXRSessionManager implements IDisposable {
         }
     }
 
+    /**
+     * Returns true if the session's backend is BabylonNative, otherwise false
+     */
+    public isNativeSession(): boolean {

```

Comment:
can be a getter, so it can be used as a readonly flag

```javascript
public get isNative(): boolean {
    return !!this._xrNavigator.xr.native;
}
```

I also think it is not the session, but the environment (this is why i removed the `Session`. But that's arguable :-)